substratevm/src/com.oracle.svm.hosted/src/com/oracle/svm/hosted/image/NativeBootImage.java

```diff
@@ -458,6 +473,12 @@ public void build(DebugContext debug, ImageHeapLayouter layouter) {
             cGlobals.writeData(rwDataBuffer, (offset, symbolName) -> defineDataSymbol(symbolName, rwDataSection, offset + RWDATA_CGLOBALS_PARTITION_OFFSET));
             defineDataSymbol(CGlobalDataInfo.CGLOBALDATA_BASE_SYMBOL_NAME, rwDataSection, RWDATA_CGLOBALS_PARTITION_OFFSET);
 
+            // if we have constructed any debug info then
+            // give the object file a chance to install it
+            if (GraalOptions.TrackNodeSourcePosition.getValue(HostedOptionValues.singleton())) {
```

Comment:
We would want to have a hosted option. e.g `GenerateDebugInfo` that would ensure TrackNodeSourcePosition is enabled whenever `GenerateDebugInfo > 0`

Reply:
Answered in another comment